### PR TITLE
feat(avatar): add user avatar as a component

### PR DESCRIPTION
**Pull request by SMahil**
### Mandatory
 - [ ] What does this PR do? 
Add User Avatar as a reusable component
- [ ] What issue/task does this PR references?
https://openshift.io/openshiftio/openshiftio/plan/detail/2528

- [ ] @mention(s) to expected reviewer(s) included
@sanbornsen @vikram-raj @rgarg1

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @SMahil for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


For more information please head over to official [documentation](http://arquillian.org/ike-prow-plugins/#_test_keeper_plugin). You can find there how to configure the plugin - for example exclude certain file types so if PR contains only them it won't be checked.

**Comment by vikram-raj** [comment]
screenshot from 2018-06-06 08-10-09

In Timeline section text moved to next line.

File: src/app/widgets/user-avatar/user-avatar.component.html [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<img class="user-assign-avatar"
+  [src]="imgSrc"
+  placement="left"
```
Comment (vikram-raj) [inline src/app/widgets/user-avatar/user-avatar.component.html]: placement should be generic. This should not be hardcoded in the component. Take placement as input.

File: src/app/widgets/user-avatar/user-avatar.component.html [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+  [tooltip]="imgTooltip"
```
Comment (vikram-raj) [inline src/app/widgets/user-avatar/user-avatar.component.html]: tooltip is not an input for img so use tooltip without the square brackets.

File: src/app/components_ngrx/work-item-detail/work-item-detail.component.html [hunk 3]
```diff
@@ -193,14 +193,14 @@
...
-                    onError="this.src='https://avatars0.githubusercontent.com/u/563119?v=3&s=20'"
-                    id="WI_details_reporter_img" />
-                </div>
+                <user-avatar 
+                  id="WI_details_reporter_img"
+                  *ngIf="workItem?.creator"
+                  [imgSrc]="workItem?.creator?.avatar + '&s=20'"
+                  [imgPlacement]="'bottom'"
+                  [imgTooltip]="workItem?.creator?.name"
+                  onError="this.src='https://avatars0.githubusercontent.com/u/563119?v=3&s=20'">
```
Comment (vikram-raj) [inline src/app/components_ngrx/work-item-detail/work-item-detail.component.html]: Remove this ```onError``` attributes.

File: src/app/components_ngrx/assignee-selector/assignee-selector.component.html [hunk 4]
```diff
@@ -21,12 +21,10 @@
...
     let-stickontop="stickontop">
-    <div class="user-avatar margin-right-10 pull-left">
-      <img
-        class="user-assign-avatar pull-left"
-        [src]="src | almAvatarSize: 20"
-        onError="this.src='https://avatars0.githubusercontent.com/u/563119?v=3&s=20'"/>
-    </div>
+    <user-avatar
+      [imgSrc]="src | almAvatarSize: 20"
+      onError="this.src='https://avatars0.githubusercontent.com/u/563119?v=3&s=20'">
```
Comment (vikram-raj) [inline src/app/components_ngrx/assignee-selector/assignee-selector.component.html]: No need to add this ``` onError ``` attributes here. Currently, it is same for all. and You already add ```onError``` in user-avatar.component.html.

File: src/app/components_ngrx/work-item-cell/work-item-cell.component.html [hunk 5]
```diff
@@ -60,19 +60,18 @@
...
-        src="{{row.creator?.avatar + '&s=23'}}"
-        onError="this.src='https://avatars0.githubusercontent.com/u/563119?v=3&s=23'" />
-</div>
-
+<user-avatar
+  *ngIf="col === 'creator'"
+  [imgSrc]= "row.creator?.avatar + '&s=23'" 
+  [imgTooltip]= "row.creator?.name"
+  [imgPlacement]="'left'"
+  onError="this.src='https://avatars0.githubusercontent.com/u/563119?v=3&s=23'">
```
Comment (vikram-raj) [inline src/app/components_ngrx/work-item-cell/work-item-cell.component.html]: Remove this ```onError``` attributes.

**Comment by rgarg1** [comment]
@jarifibrahim @Raunak1203 Pls retrigger the build. It failed due to rxjs issue, which is now resolved.

File: src/app/components/work-item-comment/work-item-comment.component.html [hunk 6]
```diff
@@ -41,16 +41,14 @@
...
-              container="body"
-              tooltip="{{comment.relationships['created-by']?.data?.attributes?.fullName}}"
-              *ngIf="comment.relationships['created-by']?.data?.attributes?.imageURL"
-              [src]="comment.relationships['created-by']?.data?.attributes?.imageURL | almAvatarSize:30"
-            />
-          </div>
+          <user-avatar
+            id="{{'comment_avatar_' + counter}}"
+            *ngIf="comment.relationships['created-by']?.data?.attributes?.imageURL"
+            container="body"
```
Comment (vikram-raj) [inline src/app/components/work-item-comment/work-item-comment.component.html]: What is this ```container="body"``` for?

Comment (SMahil) [reply src/app/components/work-item-comment/work-item-comment.component.html]: It was already there.

**Comment by SMahil** [comment]
[test]

**Comment by jarifibrahim** [comment]
[test]

**Comment by SMahil** [comment]
[test]

**Comment by nimishamukherjee** [comment]
[test]

**Comment by nimishamukherjee** [comment]
[test]

**Comment by SMahil** [comment]
@Raunak1203 I am unable to reproduce this bug
screen shot 2018-06-15 at 9 19 05 pm

**Pull request merged**